scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+        considering `nan` will filter the samples to retain only the levels 
+        of interest. This is recommended for kracken/bracken tables, but not 
+        applicable for some 16s sequences
+    abund_thresh_rough, abund_thresh_fine : float [0, 1]
+        The mean abundance threshhold for a taxonomic group to be plotted for
+        the higher level grouping (`abund_thresh_rough`) and sub grouping
+        level. This will be used in conjunction with the `group_thresh_rough`
+        and `group_thresh_fine` to determine the number of groups to be
+        included.
+    group_thresh_fine, group_thresh_rough: int, [1, 6]
+        The maximum number of taxonmic groups to display for the respective 
+        level. If `group_thresh_rough` > 6, then it will be replaced with 
+        6 because this is the maximum number of avaliable color groups.
+
+    Returns
+    -------
+    Figure
+        A 8" x 4" matplotlib figure with the area plot and legend.
+
+    Also See
+    --------
+    single_area_plot
+
+    """
+
+    # Parses the taxonomy and collapses the table
+    taxa = extract_label_array(table, tax_col, tax_delim)
+    if samples is None:
+    	samples = list(table.drop(columns=[tax_col]).columns)
+
+    # Gets the appropriate taxonomic level information to go forward
+    collapsed = level_taxonomy(table, taxa, samples, 
+                               level=np.array([fine_level]), 
+                               consider_nan=multilevel_table)
+    samples = collapsed.columns
+
+    # Gets the top taxonomic levels
+    upper_, lower_, = profile_joint_levels(collapsed, rough_level, fine_level, 
+    								       samples=samples,
+                                           lo_thresh=abund_thresh_rough, 
+                                           lo_count=min(5, group_thresh_rough),
+                                           hi_thresh=abund_thresh_fine,
+                                           hi_count=group_thresh_fine,
+                                           )
+    # Gets the colormap 
+    cmap = define_join_cmap(upper_)
+    upper_.index = upper_.index.droplevel('rough')
+    lower_.index = lower_.index.droplevel('rough')
+
+    # Plots the data
+    fig_ = plot_area(upper_.astype(float), lower_.astype(float), cmap)
+
+    return fig_
+
+
+# Sets up the main arguments for argparse.
+parser_one = argparse.ArgumentParser(
+    description=('A set of functions to generate diagnostic stacked area '
+                 'plots from metagenomic outputs.'),
+    prog=('area_plotter'),
+    )
+parser_one.add_argument('-t', '--table', 
+                    help=('The abundance table as a tsv classic biom '
+                          '(features as rows, samples as columns) containing'
+                          'absloute or relative abundance for the samples.'),
+                    required=True,
+                    )
+parser_one.add_argument('-f', '--figure',
```

Comment:
While I like that it is descriptive with `--figure`, I tend to prefer flags like `--output` or maybe  `--output-figure` to really be obvious about it being the output of the tool. It also matches the CLI of a lot of other tools better.